flink-formats/flink-avro-glue-schema-registry/src/main/java/org/apache/flink/formats/avro/glue/schema/registry/GlueSchemaRegistryAvroSchemaCoderProvider.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.formats.avro.glue.schema.registry;
+
+import org.apache.flink.formats.avro.SchemaCoder;
+
+import javax.annotation.Nonnull;
+
+import java.util.Map;
+
+/** Provider for {@link GlueSchemaRegistryAvroSchemaCoder}. */
+public class GlueSchemaRegistryAvroSchemaCoderProvider implements SchemaCoder.SchemaCoderProvider {
+    private final String transportName;
+    private final Map<String, Object> configs;
+
+    /**
+     * Constructor used by {@link GlueSchemaRegistryAvroDeserializationSchema} and {@link
+     * GlueSchemaRegistryAvroSerializationSchema}.
+     *
+     * @param transportName topic name or stream name etc.
+     * @param configs configurations for AWS Glue Schema Registry
+     */
+    public GlueSchemaRegistryAvroSchemaCoderProvider(
+            String transportName, @Nonnull Map<String, Object> configs) {
```

Comment:
As previously discussed, we should not usually annotate `@Nonnull`. Instead we should annotate `@Nullable`